script/check-size

```diff
@@ -118,137 +119,166 @@ nm_size()
-        "-DOT_BORDER_ROUTER=ON"
-        "-DOT_CHANNEL_MANAGER=ON"
-        "-DOT_CHANNEL_MONITOR=ON"
-        "-DOT_COAP=ON"
-        "-DOT_COAPS=ON"
-        "-DOT_COMMISSIONER=ON"
-        "-DOT_DATASET_UPDATER=ON"
-        "-DOT_DHCP6_CLIENT=ON"
-        "-DOT_DHCP6_SERVER=ON"
-        "-DOT_DIAGNOSTIC=ON"
-        "-DOT_DNSSD_SERVER=ON"
-        "-DOT_DNS_CLIENT=ON"
-        "-DOT_ECDSA=ON"
-        "-DOT_FULL_LOGS=ON"
-        "-DOT_JAM_DETECTION=ON"
-        "-DOT_JOINER=ON"
-        "-DOT_LINK_RAW=ON"
-        "-DOT_MAC_FILTER=ON"
-        "-DOT_MESSAGE_USE_HEAP=ON"
-        "-DOT_NETDATA_PUBLISHER=ON"
-        "-DOT_PING_SENDER=ON"
-        "-DOT_SERVICE=ON"
-        "-DOT_SLAAC=ON"
-        "-DOT_SNTP_CLIENT=ON"
-        "-DOT_SRP_CLIENT=ON"
-        "-DOT_SRP_SERVER=ON"
-        "-DOT_TIME_SYNC=ON"
-        "-DOT_UDP_FORWARD=ON"
-        "-DOT_UPTIME=ON"
-    )
+    case "$1" in
+        ftd)
+            local ot_ftd=ON
+            local ot_mtd=OFF
+            local ot_rcp=ON
+            ;;
+        mtd)
+            local ot_ftd=OFF
+            local ot_mtd=ON
+            local ot_rcp=ON
+            ;;
+        br)
+            local ot_ftd=ON
+            local ot_mtd=OFF
+            local ot_rcp=OFF
+            ;;
+        *)
+            exit 128
+            ;;
+    esac
+
+    case "$2" in
+        new)
+            local sha=${OT_SHA_NEW}
+            ;;
+        old)
+            local sha=${OT_SHA_OLD}
+            ;;
+        *)
+            exit 128
+            ;;
+    esac
 
-    local thread_version=$1
-
-    if [[ ${thread_version} != "1.1" ]]; then
-        options+=(
-            "-DOT_THREAD_VERSION=1.3"
-            "-DOT_BACKBONE_ROUTER=ON"
-            "-DOT_DUA=ON"
-            "-DOT_MLR=ON"
-            "-DOT_CSL_RECEIVER=ON"
-            "-DOT_LINK_METRICS_INITIATOR=ON"
-            "-DOT_LINK_METRICS_SUBJECT=ON"
-        )
+    local folder="$1_$2"
+    local config_name="ot-core-config-check-size-$1.h"
+    local config_file="../examples/config/${config_name}"
+
+    mkdir -p "${OT_TMP_DIR}/${folder}"
+    script/git-tool clone https://github.com/openthread/ot-nrf528xx.git "${OT_TMP_DIR}/${folder}"
+    rm -rf "${OT_TMP_DIR}/${folder}/openthread/*" # replace openthread submodule with latest commit
+    git archive "${sha}" | tar x -C "${OT_TMP_DIR}/${folder}/openthread"
+
+    if [ ! -e "${OT_TMP_DIR}/${folder}/openthread/examples/config/${config_name}" ]; then
+        # Check if the the config headers are not present, copy from
+        # the main sha.
+        case "$1" in
+            br)
+                rm -rf "${OT_TMP_DIR}/${folder}/openthread/*"
+                git archive "${OT_SHA_NEW}" | tar x -C "${OT_TMP_DIR}/${folder}/openthread"
+                ;;
+            *)
+                mkdir -p "${OT_TMP_DIR}/${folder}/openthread/examples/config"
+                cp "./examples/config/${config_name}" "${OT_TMP_DIR}/${folder}/openthread/examples/config"
+                ;;
+        esac
```

Comment:
This `if` above is mainly intended for smooth transition between old model and new model (since the code with "old sha" won't have the same config headers). We can remove this in a future PR.